### PR TITLE
Add eval details template for deny-by-default rule.

### PR DESCRIPTION
**Pull request by blkt**
# Summary

This change adds a template used to produce evaluation details for the `rego/deny-by-default` evaluation engine.

This template allows the user defining the rule to output a `message` variable with a custom message to show when the evaluation fails.

Fixes stacklok/minder-stories#42

## Change Type

- [ ] Bug fix (resolves an issue without affecting existing features)
- [X] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

Unit and manual tested.

# Review Checklist:

- [X] Reviewed my own code for quality and clarity.
- [X] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [X] Checked that related changes are merged.

### CODE REVIEW

File: internal/engine/rtengine/engine.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (r *RuleTypeEngine) Eval(
 
 	// Process evaluation
 	logger.Info().Msg("entity evaluation - evaluation started")
-	err := r.ruleEvaluator.Eval(ctx, params.GetRule().Def, result)
+	err := r.ruleEvaluator.Eval(ctx, params.GetRule().Def, inf.Entity, result)
```
Comment (blkt) [inline internal/engine/rtengine/engine.go:180]: I could turn this into the same pattern as the one used [here](https://github.com/stacklok/minder/pull/4584) to limit the changes across the code base, let me know what you folks prefer.

File: internal/engine/eval/templates/regoDenyByDefaultTemplate.tmpl [hunk 2]
```diff
@@ -0,0 +1 @@
+Evaluation {{ .status }}: {{ .message }}{{ with .entityName }} for entity {{ . }}{{ end }}
```
Comment (eleftherias) [inline internal/engine/eval/templates/regoDenyByDefaultTemplate.tmpl:1]: Based on https://github.com/stacklok/minder-stories/issues/42#issuecomment-2394190172 should this be `Evaluation {{ .status }}: {{ .message }}{{ with .entityName }} for {{ . }}{{ end }}`?

Comment (eleftherias) [reply internal/engine/eval/templates/regoDenyByDefaultTemplate.tmpl:1]: Also, should we remove the `Evaluation {{ .status }}:` prefix? I think all the places that would use this already show the status separately from the details.

Comment (blkt) [reply internal/engine/eval/templates/regoDenyByDefaultTemplate.tmpl:1]: Yes, I might have prematurely optimized for the other cases (non `failed` ones).
I'll remove that.

**Review by eleftherias — APPROVED** [review]

**Pull request merged**